src/net/tcp_stream.rs

```diff
@@ -311,7 +321,8 @@ impl Read for TcpStream {
             Ok(nread_or_error_id as usize)
         } else {
             let lunatic_error = LunaticError::from(nread_or_error_id);
```

Comment:
If a `tcp_read` times out, the error_id is not going to point to a valid error object and it's not safe to create a `LunaticError` from it.

I didn't test this, but I assume that a process would fail if trying to drop a `LunaticError` with an invalid reference.

Reply:
Should I create a new entry in the LunaticError enum then? Like TimedOut? Or should I just return a `Error::from(ErrorKind::TimedOut, "Write call timed out")`?

Reply:
I'm ok with both, whatever feels better to you.

Reply:
I chose to send a `str` with a description of the error because it's not really a lunatic error, but a tcp stream error and I'd like to keep the error mapping close to the logic where it's called from.
I also added a peek function to the lib because I forgot to expose it previously.